### DISCUSSION TITLE
silly question, I can't print the sprite name....

### DISCUSSION THREAD

**Discussion by rudiHammad** [axmolengine/axmol#1679] [Q&A] [answered]
I am sure it must be something stupid by I am trying to print the name of a sprite and I don't know how

`   auto testSprite = Sprite::create();
    testSprite->setName("fooo");
    log("%s", testSprite->getName());`

This prints simbols. and getName() doesn't have .c_str(), so I am not sure how I can't figure out something so simple.

thx

**Comment by rh101** [comment]
`getName()` returns `std::string_view`. if you want the raw string then you need to use the `std::string_view::data()` method.  In this case though, it's best to cast it to a `std::string` first and then just use `c_str()`, to ensure it is null terminated.

Use `std::string(testSprite.getName()).c_str()`.

**Comment by rudiHammad** [comment]
of course... thanks.
what use then should we give to std::string_view? In theory it is preferable to use it instead of std::string because it's light weight right?

**Reply by rh101** [reply, marked as answer]
> what use then should we give to std::string_view?

Sorry, I don't understand what you mean by that.  

> In theory it is preferable to use it instead of std::string because it's light weight right?

It's not theory, but you're right that it's "light weight", since it's just a read-only reference to data, and doesn't actually hold a copy of the data.  Plenty of info online about it.

In your specific use-case above, you're attempting to use a function (`log`) that accepts a null terminated character sequence, which has nothing to do with `std::string` or `std::string_view`.  The need to convert the `string_view` to a `string` first is because the `string_view` does not have a null terminator at the end of the text, and if you don't ensure there is a null terminator (which a `std::string` does have), then bad things can happen when using the `log` method.

**Comment by rudiHammad** [comment]
Oh never mind what I meant, I understand now with your explanation.
Cool. I had no idea how the log function worked. Thanks

**Comment by iAndyHD3** [comment]
The problem here is the `log` function, which is just a wrapper for printf. Prefer using C++ printing such as std::cout or fmt::print that do support std::string_view

**Reply by halx99** [reply]
planed to replace ax::log with fmt in the future release
